src/cdk-experimental/popover-edit/table-directives.ts

```diff
@@ -0,0 +1,253 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import {fromEvent} from 'rxjs';
+import {debounceTime, filter, map, mapTo, takeUntil} from 'rxjs/operators';
+import {
+  AfterViewInit,
+  Directive,
+  ElementRef,
+  EmbeddedViewRef,
+  Input,
+  NgZone,
+  OnDestroy,
+  TemplateRef,
+  ViewContainerRef
+} from '@angular/core';
+import {FocusTrap, FocusTrapFactory} from '@angular/cdk/a11y';
+import {Overlay, OverlayRef} from '@angular/cdk/overlay';
+import {TemplatePortal} from '@angular/cdk/portal';
+import {
+  closest,
+  Destroyable,
+  DEFAULT_HOVER_DELAY_MS,
+  CELL_SELECTOR,
+  ROW_SELECTOR,
+  EDIT_PANE_CLASS,
+  EDIT_PANE_SELECTOR
+} from './common';
+import {EditEvents} from './edit-events';
+import {PopoverEditPositionStrategyFactory} from './popover-edit-position-strategy-factory';
+
+/**
+ * A directive that must be attached to enable editability on a table.
+ * It is responsible for setting up delegated event handlers and providing the
+ * EditEvents service for use by the other edit directives.
+ */
+@Directive({
+  selector: 'table[editable], cdk-table[editable], mat-table[editable]',
+  providers: [EditEvents],
+})
+export class CdkEditable extends Destroyable implements AfterViewInit {
+  constructor(
+      protected readonly elementRef: ElementRef,
+      protected readonly events: EditEvents,
+      protected readonly ngZone: NgZone) {
+    super();
+  }
+
+  ngAfterViewInit() {
```

Comment:
Can this be done in the constructor

Reply:
Is elementRef hydrated that early?

Reply:
According to this, it's ready afterViewInit

https://stackoverflow.com/questions/40660517/what-is-the-state-of-elementref-when-its-available-in-components-constructor